testing/conftest.py

```diff
@@ -17,92 +15,105 @@
+        "--headless",
+        action="store_true",
+        default=False,
+        help="Run tests in headless mode (no browser window) default its run in headed mode.",
+    )
+    parser.addoption(
+        "--slowmo",
+        action="store",
+        type=int,
+        default=0,
+        help="Slow down Playwright operations by specified milliseconds (default: 0, no slowdown)",
     )
     parser.addoption(
         "--pf-version",
         help="Patternfly Version",
         choices=("v6", "v5"),
         default="v6",
     )
-
     parser.addoption("--force-host", default=None, help="force a selenium hostname")
 
 
 @pytest.fixture(scope="session")
 def browser_name(pytestconfig):
-    return os.environ.get("BROWSER") or pytestconfig.getoption("--browser-name")
+    return os.environ.get("BROWSER") or pytestconfig.getoption("--browser")
+
+
+@pytest.fixture(scope="session")
+def headless_mode(request):
+    """Determine if tests should run in headless mode."""
+    if request.config.getoption("--headless"):
+        return True
+    return False
+
+
+@pytest.fixture(scope="session")
+def slowmo_delay(request):
+    """Get slowmo delay from command line argument."""
+    return request.config.getoption("--slowmo")
 
 
 @pytest.fixture(scope="session")
-def pf_version(pytestconfig):
+def pf_version(pytestconfig) -> str:
     return os.environ.get("PF-VERSION") or pytestconfig.getoption("--pf-version")
 
 
 @pytest.fixture(scope="session")
-def selenium_url(pytestconfig, browser_name, worker_id):
-    forced_host = pytestconfig.getoption("--force-host")
-    if forced_host is None:
-        oktet = 1 if worker_id == "master" else int(worker_id.lstrip("gw")) + 1
-        host = f"127.0.0.{oktet}"
-        ps = subprocess.run(
-            [
-                "podman",
-                "run",
-                "--rm",
-                "-d",
-                "-p",
-                f"{host}:4444:4444",
-                "-p",
-                f"{host}:7900:7900",
-                "-e",
-                "SE_VNC_NO_PASSWORD=1",
-                "-e",
-                "SE_SCREEN_HEIGHT=1080",
-                "-e",
-                "SE_SCREEN_WIDTH=1920",
-                "--shm-size=2g",
-                f"docker.io/selenium/standalone-{browser_name}:latest",
-            ],
-            stdout=subprocess.PIPE,
-            check=False,
+def playwright_browser_instance(
+    browser_name: str, headless_mode: bool, slowmo_delay: int
+) -> PlaywrightBrowser:
+    """Launches a Playwright browser instance."""
+    with sync_playwright() as p:
+        # Select browser based on command line argument (default to chromium)
+        if browser_name == "firefox":
+            py_browser = p.firefox.launch(headless=headless_mode, slow_mo=slowmo_delay)
+        else:
+            py_browser = p.chromium.launch(headless=headless_mode, slow_mo=slowmo_delay)
+
+        slowmo_info = f" with {slowmo_delay}ms slowmo" if slowmo_delay > 0 else ""
+        print(
```

Comment:
https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.hookspec.pytest_report_header

Consider moving the reflection of these options into the report header hook.  The config is available there directly.